### PR TITLE
Add support for clipping side in clip_overflow_text

### LINKED ISSUE #308: Clipping overflowing text from the left.
The current implementation of `clip_overflow` in `fury.ui` allows us to clip overflowing text from the right end for a given width. It will be helpful to have a feature that can clip overflowing text from the left end as well. The overflowing text must be replaced by `...`. This feature will be helpful while developing `FileDialog2D`.

For e.g 
if `extraordinary` overflows for a given width then it must be replaced by `...ordinary` where `...ordinary` is the perfect fit for that width.

### PR DESCRIPTION
**Pull request by tushar5526**
Closes #308 
If we have to clip from left, we reverse the string before processing it and then reverse it again before returning it.

### CODE REVIEW

**Comment by tushar5526** [comment]
@nibba Should I debug a warning if wrong argument values are passed or throw an error ? ( if clip is not equal to "left" or "right")

File: fury/ui.py [hunk 1]
```diff
@@ -42,6 +45,9 @@ def clip_overflow(textblock, width):
         textblock.have_bg = prev_bg
         return original_str
 
+    if side == 'left':
```
Comment (skoudoro) [inline fury/ui.py:48]: `side.lower()`  instead (just in case)

File: fury/ui.py [hunk 2]
```diff
@@ -53,6 +59,8 @@ def clip_overflow(textblock, width):
         if mid_ptr == (start_ptr + end_ptr)//2 or\
            textblock.size[0] == width:
             textblock.have_bg = prev_bg
+            if side == 'left':
```
Comment (skoudoro) [inline fury/ui.py:62]: `side.lower()`  instead (just in case)

**Review by skoudoro — COMMENTED** [review]
Hi @tushar5526,

Nice trick. Could you update the [tests](https://github.com/fury-gl/fury/blob/master/fury/tests/test_ui.py#L1571) and add this case.

Thanks!

**Comment by skoudoro** [comment]
> Should I debug a warning if wrong argument values are passed or throw an error ? ( if side != "left" or "right")

yes, if wrong argument, raise a `ValueError`

File: fury/tests/test_ui.py [hunk 3]
```diff
@@ -1586,3 +1586,15 @@ def test_clip_overflow():
...
+    ui.clip_overflow(text, rectangle.size[0], 'left')
+    npt.assert_equal("Hello", text.message)
+
+    text.message = "Hello wassup"
+    ui.clip_overflow(text, rectangle.size[0], 'left')
+    npt.assert_equal("...lo wassup", text.message)
+
+    text.message = "A very very long message to clip text overflow"
+    ui.clip_overflow(text, rectangle.size[0], 'left')
+    npt.assert_equal("... overflow", text.message)
```
Comment (skoudoro) [inline fury/tests/test_ui.py:1600]: Can you add 2 more tests: 
- one with uppercase:  `ui.clip_overflow(text, rectangle.size[0],  'LeFT')`
- check that the error is raised:  `np.assert_raise(ValueError, ui.clip_overflow, text, rectangle.size[0], 'middle')`

File: fury/ui.py [hunk 4]
```diff
@@ -26,12 +26,19 @@ def clip_overflow(textblock, width):
         The textblock object whose text needs to be clipped.
     width : int
         Required width of the clipped text.
+    side : string
```
Comment (skoudoro) [inline fury/ui.py:29]: `side : str, optional` instead

File: fury/ui.py [hunk 5]
```diff
@@ -26,12 +26,19 @@ def clip_overflow(textblock, width):
...
+        Clips the overflowing text according to side.
+        It takes values "left" or "right".
 
     Returns
     -------
     clipped text : str
         Clipped version of the text.
     """
+    side = side.lower()
+    if not (side == 'left' or side == 'right'):
```
Comment (skoudoro) [inline fury/ui.py:39]: `if side not in ['left', 'right']`  instead

**Review by skoudoro — COMMENTED** [review]
Thank you for the update @tushar5526.

Here some last comment

**Comment by tushar5526** [comment]
Sure @skoudoro

File: fury/tests/test_ui.py [hunk 6]
```diff
@@ -1586,3 +1586,25 @@ def test_clip_overflow():
...
+
+    text.message = "A very very long message to clip text overflow"
+    ui.clip_overflow(text, rectangle.size[0], 'LeFT')
+    npt.assert_equal("... overflow", text.message)
+
+    text.message = "A very very long message to clip text overflow"
+    ui.clip_overflow(text, rectangle.size[0], 'RigHT')
+    npt.assert_equal("A very ve...", text.message)
+
+    npt.assert_raises(ValueError, ui.clip_overflow, text, rectangle.size[0], 'middle')
```
Comment (skoudoro) [inline fury/tests/test_ui.py:1610]: [pep8](https://github.com/fury-gl/fury/pull/337#issuecomment-731074422): line too long

**Review by skoudoro — APPROVED** [review]
Apart from the small pep8, All good for me! Thank you @tushar5526.

I will wait for the @Nibba2018 review and then we can merge it.

**Comment by Nibba2018** [comment]
That's a smart fix @tushar5526 !. Apart form that PEP8 line everything else looks good to me. :rocket:

**Review by Nibba2018 — APPROVED** [review]

**Comment by tushar5526** [comment]
Fixed the PEP 8

**Comment by skoudoro** [comment]
Thanks @Nibba2018 for the review and Thanks @tushar5526 for this feature! merging

**Pull request merged**